ci: failure in p2p_addr_selfannouncement.py

https://github.com/bitcoin/bitcoin/actions/runs/20445863729/job/58749036885#step:9:3498:
```bash
 test  2025-12-22T23:47:48.293698Z TestFramework.p2p (DEBUG): Received message from 127.0.0.1:12164: msg_pong(nonce=00000000) 
 test  2025-12-22T23:47:48.293794Z TestFramework.p2p (DEBUG): Received message from 127.0.0.1:12164: msg_pong(nonce=00000006) 
 test  2025-12-22T23:47:48.344658Z TestFramework (ERROR): Unexpected exception 
                                   Traceback (most recent call last):
                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/test/functional/test_framework/test_framework.py", line 142, in main
                                       self.run_test()
                                       ~~~~~~~~~~~~~^^
                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch/build-aarch64-apple-darwin24.6.0/test/functional/p2p_addr_selfannouncement.py", line 67, in run_test
                                       self.self_announcement_test(outbound=False, addrv2=True)
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch/build-aarch64-apple-darwin24.6.0/test/functional/p2p_addr_selfannouncement.py", line 135, in self_announcement_test
                                       assert_equal(addr_receiver.self_announcements_received, last_self_announcements_received + 1)
                                       ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/test/functional/test_framework/util.py", line 80, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(4 == 5)
 test  2025-12-22T23:47:48.347601Z TestFramework (DEBUG): Closing down network thread 
```

got this with the diff above 

```
2025-12-31T12:39:50.147708Z TestFramework (INFO): Check that we get an initial self-announcement when connecting to a node and sending a GETADDR (inbound, addrv1)
>>> HANDLE_ADDR_MESSAGE called, expected.time=1767184790
2025-12-31T12:39:50.304489Z TestFramework (INFO): Check that we get more self-announcements sometime later (inbound, addrv1)
BEFORE bumpmocktime()
>>> HANDLE_ADDR_MESSAGE called, expected.time=1767184790
Self-announcement timestamp mismatch: got=1768912790 expected=1767184790
AFTER bumpmocktime(), before setting expected.time
2025-12-31T12:39:50.459889Z TestFramework (ERROR): Unexpected exception
Traceback (most recent call last):
  File "/home/ubuntu/Projects/bitcoin/test/functional/test_framework/test_framework.py", line 142, in main
    self.run_test()
  File "/home/ubuntu/Projects/bitcoin/./build/test/functional/p2p_addr_selfannouncement.py", line 69, in run_test
    self.self_announcement_test(outbound=False, addrv2=False)
  File "/home/ubuntu/Projects/bitcoin/./build/test/functional/p2p_addr_selfannouncement.py", line 141, in self_announcement_test
    assert_equal(addr_receiver.self_announcements_received, last_self_announcements_received + 1)
  File "/home/ubuntu/Projects/bitcoin/test/functional/test_framework/util.py", line 80, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(1 == 2)
```

On NetBSD 9.4, https://github.com/hebasto/bitcoin-core-nightly/actions/runs/20619499569/job/59218560248:
```
182/282 - p2p_addr_selfannouncement.py failed, Duration: 6 s

stdout:
2025-12-31T14:10:01.054524Z TestFramework (INFO): PRNG seed is: 279208991052383101
2025-12-31T14:10:01.060063Z TestFramework (INFO): Initializing test directory /home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test_runner_₿_🏃_20251231_135038/p2p_addr_selfannouncement_96
2025-12-31T14:10:03.459488Z TestFramework (INFO): Test that the node does an address self-announcement to inbound connections (addrv1)
2025-12-31T14:10:03.472186Z TestFramework (INFO): Check that we get an initial self-announcement when connecting to a node and sending a GETADDR (inbound, addrv1)
2025-12-31T14:10:04.247264Z TestFramework (INFO): Check that we get more self-announcements sometime later (inbound, addrv1)
2025-12-31T14:10:04.961413Z TestFramework (INFO): Test that the node does an address self-announcement to inbound connections (addrv2)
2025-12-31T14:10:04.963778Z TestFramework (INFO): Check that we get an initial self-announcement when connecting to a node and sending a GETADDR (inbound, addrv2)
2025-12-31T14:10:05.115558Z TestFramework (INFO): Check that we get more self-announcements sometime later (inbound, addrv2)
2025-12-31T14:10:05.739087Z TestFramework (ERROR): Unexpected exception
Traceback (most recent call last):
  File "/home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test/functional/test_framework/test_framework.py", line 142, in main
    self.run_test()
    ~~~~~~~~~~~~~^^
  File "/home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/build/test/functional/p2p_addr_selfannouncement.py", line 67, in run_test
    self.self_announcement_test(outbound=False, addrv2=True)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/build/test/functional/p2p_addr_selfannouncement.py", line 135, in self_announcement_test
    assert_equal(addr_receiver.self_announcements_received, last_self_announcements_received + 1)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test/functional/test_framework/util.py", line 80, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(5 == 6)
2025-12-31T14:10:05.745570Z TestFramework (INFO): Not stopping nodes as test failed. The dangling processes will be cleaned up later.
2025-12-31T14:10:05.745741Z TestFramework (WARNING): Not cleaning up dir /home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test_runner_₿_🏃_20251231_135038/p2p_addr_selfannouncement_96
2025-12-31T14:10:05.745854Z TestFramework (ERROR): Test failed. Test logging available at /home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test_runner_₿_🏃_20251231_135038/p2p_addr_selfannouncement_96/test_framework.log
2025-12-31T14:10:05.746147Z TestFramework (ERROR): 
2025-12-31T14:10:05.746352Z TestFramework (ERROR): Hint: Call /home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test/functional/combine_logs.py '/home/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test_runner_₿_🏃_20251231_135038/p2p_addr_selfannouncement_96' to consolidate all logs
2025-12-31T14:10:05.746466Z TestFramework (ERROR): 
2025-12-31T14:10:05.746574Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2025-12-31T14:10:05.746787Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2025-12-31T14:10:05.746885Z TestFramework (ERROR): 


stderr:
[node 0] Cleaning up leftover process

```

--- a/test/functional/p2p_addr_selfannouncement.py
+++ b/test/functional/p2p_addr_selfannouncement.py
@@ -128,8 +128,8 @@ class AddrSelfAnnouncementTest(BitcoinTestFramework):
                 # self-announcements are sent on an exponential distribution with mean interval of 24h.
                 # Setting the mocktime 20d forward gives a probability of (1 - e^-(480/24)) that
                 # the event will occur (i.e. this fails once in ~500 million repeats).
+                addr_receiver.expected.time = self.nodes[0].mocktime + 20 * ONE_DAY
                 self.nodes[0].bumpmocktime(20 * ONE_DAY)
-                addr_receiver.expected.time = self.nodes[0].mocktime
                 addr_receiver.sync_with_ping()
 
             assert_equal(addr_receiver.self_announcements_received, last_self_announcements_received + 1)

